common/hyperscaler/rules/rule.go

```diff
@@ -0,0 +1,224 @@
+package rules
+
+import (
+	"fmt"
+	"sort"
+)
+
+type Labels struct {
+	Labels []string
+}
+
+type Rule struct {
+	Plan                           string
+	PlatformRegion                 string
+	HyperscalerRegion              string
+	EuAccess                       bool
+	Shared                         bool
+	Labels                         map[string]string
+	ContainsInputAttributes        bool
+	ContainsOutputAttributes       bool
+	hyperscalerNameMappingFunction func(string) string
+}
+
+func NewRule() *Rule {
+	return &Rule{
+		hyperscalerNameMappingFunction: getHyperscalerName,
+		Labels:                         make(map[string]string),
+	}
+}
+
+type MatchableAttributes struct {
+	Plan              string `json:"plan"`
+	PlatformRegion    string `json:"platformRegion"`
+	HyperscalerRegion string `json:"hyperscalerRegion"`
+}
+
+func (r *Rule) CalculateLabels() map[string]string {
+	return r.CalculateLabelsWith(getHyperscalerName(r.Plan))
+}
+
+func (r *Rule) CalculateLabelsWith(hyperscalerName string) map[string]string {
+	for _, attr := range AllAttributes {
+		if attr.Getter(r) != "" {
+			r.Labels = attr.ApplyLabel(r, r.Labels)
+		}
+	}
+
+	return r.Labels
+}
+
+func getHyperscalerName(plan string) (result string) {
+	if plan == "aws" || plan == "gcp" || plan == "azure" || plan == "azure_lite" {
+		return plan
+	} else if plan == "trial" {
+		return "aws"
+	} else if plan == "free" {
+		return "aws/azure"
```

Comment:
`aws/azure` is not proper hyperscaler name.
And hardcoding all this relations here we need to update this function every time we add a plan.

Reply:
Just a temporary implementation (for cli) that will be replaced with hyperscaler_type.go

Reply:
When and how this be replaced?
In the scope of this PR?

Reply:
I planned to add it in https://github.com/kyma-project/kyma-environment-broker/pull/1796